bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/GsonMapper.java

```diff
@@ -0,0 +1,319 @@
+            Boolean bool;
+            if (value == null) {
+                bool = null;
+            } else if (value instanceof Boolean) {
+                bool = (Boolean) value;
+            } else if (value instanceof Number) {
+                bool = ((Number) value).longValue() != 0;
+            } else if (value instanceof String) {
+                bool = parseBoolean((String) value);
+            } else {
+                logger.warn("Cannot parse boolean from [{}]", value);
+                bool = null;
+            }
+            return new DeviceProperty.BooleanDeviceProperty(name, readOnly, direction, dataUpdatedAt, productName,
+                    displayName, bool, properties);
+        }
+        if ("integer".equalsIgnoreCase(baseType)) {
+            Long longValue;
+            if (value == null) {
+                longValue = null;
+            } else if (value instanceof Long) {
+                longValue = (Long) value;
+            } else if (value instanceof Number) {
+                longValue = ((Number) value).longValue();
+            } else if (value instanceof String string) {
+                try {
+                    longValue = parseLong(string);
+                } catch (NumberFormatException ex) {
+                    logger.warn("Cannot parse long from [{}]", value, ex);
+                    longValue = null;
+                }
+            } else {
+                logger.warn("Cannot parse long from [{}]", value);
+                longValue = null;
+            }
+            return new DeviceProperty.LongDeviceProperty(name, readOnly, direction, dataUpdatedAt, productName,
+                    displayName, longValue, properties);
+        }
+        var string = value != null ? value.toString() : null;
+        return new DeviceProperty.StringDeviceProperty(name, readOnly, direction, dataUpdatedAt, productName,
+                displayName, string, properties);
+    }
+
+    private String findOrNull(Map<?, ?> map, String name) {
+        if (!map.containsKey(name)) {
+            return null;
+        }
+        return (String) map.get(name);
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private Boolean findBoolOrNull(Map<?, ?> map, String name) {
+        if (!map.containsKey(name)) {
+            return null;
+        }
+        var value = map.get(name);
+        if (value == null) {
+            return null;
+        }
+        if (value instanceof Boolean bool) {
+            return bool;
+        }
+        if (value instanceof String string) {
+            return parseBoolean(string);
+        }
+        return null;
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private Object findObjectOrNull(Map<?, ?> map, String name) {
+        if (!map.containsKey(name)) {
+            return null;
+        }
+        return map.get(name);
+    }
+
+    public String datapointParam(Object value) {
+        return gson.toJson(singletonMap("datapoint", singletonMap("value", value)));
```

Comment:
```suggestion
        return gson.toJson(Map.of("datapoint", Map.of("value", value)));
```

Reply:
Is there any difference between `singletonMap` and `Map.of`?